api/v1beta1/hypershiftserviceconfig_types.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HypershiftServiceConfigSpec defines the desired state of HypershiftServiceConfig.
+type HypershiftServiceConfigSpec struct {
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Agent Service Configuration"
+	AgentConfig AgentServiceConfigSpec `json:"agentConfig"`
```

Comment:
Don't we want to use composition instead? I.e. embed the current AgentServiceConfigSpec struct? It would make the API exactly the same as ASC just with the new KubeconfigSecretRef.
E.g.
```
type HypershiftServiceConfigSpec struct {
	AgentServiceConfigSpec `json:",inline"`

	KubeconfigSecretRef *corev1.LocalObjectReference `json:"kubeconfigSecretRef,omitempty"`
}
```